Mathlib/LinearAlgebra/AffineSpace/AffineMap.lean

```diff
@@ -563,6 +563,17 @@ theorem lineMap_vsub_lineMap (p₁ p₂ p₃ p₄ : P1) (c : k) :
     lineMap (lineMap p₀ p₁ c) p₁ d = lineMap p₀ p₁ (1 - (1 - d) * (1 - c)) := by
   simp_rw [lineMap_apply_one_sub, ← lineMap_apply_one_sub p₁, lineMap_lineMap_right]
 
+lemma lineMap_monotone [LinearOrder k] [IsOrderedRing k] {p₀ p₁ : k} (h : p₀ ≤ p₁) :
+    Monotone (lineMap (k := k) p₀ p₁) := by
+  intro x y hxy
+  have : 0 ≤ p₁ - p₀ := sub_nonneg_of_le h
+  simpa [lineMap] using by gcongr
+
+lemma lineMap_antitone [LinearOrder k] [IsOrderedRing k] {p₀ p₁ : k} (h : p₁ ≤ p₀) :
```

Comment:
Does this generalize conveniently to ordered modules? For instance, it certainly holds for the lineMap along the real line of the complex numbers

Reply:
I'm happy to accept "no, not conveniently" in this case then :). Perhaps it makes sense to remark in a docstring that this is the special case of `lineMap` over the 1D space that is the ring.

Reply:
> I found `OrderedSMul`, but the file says that it is useless and we should remove i

The file should also say to use `PosSMulMono` and friends 😉